tests/rust/showcase/main.rs

```diff
@@ -0,0 +1,41 @@
+include!("spec.rs");
+
+use std::iter::FromIterator;
+
+fn main() {
+    let customer = Customer {
+        name: "somename".to_owned(),
+        id: -23,
+        net_worth: 0.123456,
+        join_date: ::chrono::prelude::Utc::now(),
+        birthday: ::chrono::prelude::Utc::now().naive_utc().date(),
+        is_vip: true,
+        favorite_color: Color::Blue,
+        aliases: vec!["SomeName", "Some Name"]
+            .into_iter()
+            .map(|s| s.to_owned())
+            .collect(),
+        coords: (23, 42),
+        email: Some("mail".to_owned()),
+        bets: std::collections::HashMap::from_iter(
+            vec![("foo", 1.234), ("bar", -0.123)]
+                .into_iter()
+                .map(|(k, v)| (k.to_owned(), v)),
+        ),
+    };
+
+    let serialized = serde_json::to_string(&customer).expect("serialize customer");
+    let deserialized: Customer = serde_json::from_str(&serialized).expect("deserialize customer");
+
+    // FIXME use assert_eq on struct once we auto-derive PartialEq for structs
```

Comment:
I think it's time we collected all FIXMEs as issues instead and kept the code clean?